### PR TITLE
 Added extra time for retries and added logic to only check condor for incomplete jobs

### PR DESCRIPTION
**Pull request by bio-boris**
I added the following logic:
1) Don't check condor if job status is complete in UJS (I thought we always wanted to know the condor state, but this was a wrong assumption)
2) Increase retries again to fight the intermittent condor_q not working problem with condor
3) The retries will not affect narratives with awejobs that have completed, since condor is no longer being checked.
4) Only narratives with AweJobs that are still running at the time of opening the narrative will be slow
5) The fix for 4 is to add a META tag to UJS or add a field to mongo and then write logic based on that to not check the status in condor if it doesn't have the condor flag. (Other fix is to wait until no more awe jobs are running anymore)

### CODE REVIEW

File: src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java [hunk 1]
```diff
@@ -823,20 +824,27 @@ public static JobState checkJobCondor(String jobId, AuthToken authPart,
...
 
 			String stage = jobStatus.getE2();
 			if (stage != null && stage.equals("started")) {
 				returnVal.setJobState(APP_STATE_STARTED);
+				//Faking it here
+				returnVal.getAdditionalProperties().put("awe_job_state", APP_STATE_STARTED);
+				returnVal.getAdditionalProperties().put("condor_job_state", APP_STATE_STARTED);
 			} else {
 				returnVal.setJobState(APP_STATE_QUEUED);
+				//Check to see if it is really queued
```
Comment (bio-boris) [inline src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java]: Look no condor! Anyone have thoughts about this?

File: src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java [hunk 2]
```diff
@@ -823,30 +824,37 @@ public static JobState checkJobCondor(String jobId, AuthToken authPart,
...
 
 			String stage = jobStatus.getE2();
 			if (stage != null && stage.equals("started")) {
 				returnVal.setJobState(APP_STATE_STARTED);
+				//Faking it here
+				returnVal.getAdditionalProperties().put("awe_job_state", APP_STATE_STARTED);
+				returnVal.getAdditionalProperties().put("condor_job_state", APP_STATE_STARTED);
 			} else {
 				returnVal.setJobState(APP_STATE_QUEUED);
-				try {
```
Comment (bio-boris) [inline src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java]: No more condor_q calls

File: src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java [hunk 3]
```diff
@@ -106,9 +106,14 @@ public File run(
             }
             cntCmd = cntCmd.withEnv(envVarList.toArray(new String[envVarList.size()]));
 
-            if(! System.getenv("KB_DOCKER_NETWORK").equals("")) {
-                cntCmd.withNetworkMode(System.getenv("KB_DOCKER_NETWORK"));
-            }
+//            try {
```
Comment (kkellerlbl) [inline src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java]: Are we sure this is no longer required?

Comment (bio-boris) [reply src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java]: Yes this  is for mini kb

Comment (kkellerlbl) [reply src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java]: Looks good.

Comment (scanon) [reply src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java]: Do you mean this was added for mini kb?  I think I use this when I run at NERSC.  I don't think we should just drop this.

Comment (bio-boris) [reply src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java]: @sychan We have a docker network called mini_kb inside mini_kb. This code is there to be uncommented when doing testing in mini_kb, and I add the ENV variable called KB_DOCKER_NETWORK to the localmethodrunner.sh file.

Comment (sychan) [reply src/us/kbase/narrativejobservice/sdkjobs/DockerRunner.java]: I see - so you're just checking to see if it is running in a docker network at all. Depending on future deployment models, it is not out of the question that hosts might run in a docker overlay network. But we can burn that bridge when we come to it, I suppose.

**Pull request merged**